### PR TITLE
Switch client config from YAML to JSON

### PR DESCRIPTION
**Pull request by geigerj**
This corresponds to the change in packaging from providing the old
YAML client config to the new JSON client config, which includes some
minor structural and naming changes to the configuration format. Most
notably, the config is now intended to be end-user-editable, so page
descriptors and bundle descriptors have been removed from the config.

Also adds support for specifying limits in the bundle options,
although they are currently ignored by the implementations.

### CODE REVIEW

File: google/gax/__init__.py [hunk 1]
```diff
@@ -269,17 +284,29 @@ def __new__(cls,
...
+             elements in the repeated field of the bundle; if adding a request
+             to a bundle would exceed this value, the bundle is sent and the new
+             request is added to a fresh bundle. It is invalid for a single
+             request to exceed this limit.
+           request_byte_threshold: the bundled request will be sent once the
+             count of bytes in the request reaches this value. Note that this
+             value is pessimistically approximated by summing the bytesizes of
+             the elements in the repeated field, with a buffer applied to
+             compensate for the corresponding under-approximation.
+           element_count_limit: represents a hard limit on the number of
```
Comment (tbetbetbe) [inline google/gax/__init__.py]: element_count_limit occurs twice

Comment (geigerj) [inline google/gax/__init__.py]: Fixed.

**Comment by tbetbetbe** [comment]
Please fix the doc build

**Comment by tbetbetbe** [comment]
LGTM

**Comment by geigerj** [comment]
Fixed with an update to tox.ini -- the rtd theme needed to be specified explicitly as a dependency with the newest version of Sphinx

**Pull request merged**